### PR TITLE
Ccs 3510 back button bug fix

### PR DESCRIPTION
**Pull request by grantsydney**
We now have the ability to press the back button from Product Details to return to Product Listings

- refactored `productDetails.tsx` and `productListing.tsx` so they are now functional components
- created `ProductContext` so `productDetails.tsx` and `productListing.tsx` can access all product info

### CODE REVIEW

File: pantheon-bundle/frontend/src/app/productDetails.test.tsx [hunk 1]
```diff
@@ -1,137 +1,112 @@
...
+  ['sling:resourceType']: "pantheon/product",
+  urlFragment: "testurl",
+}];
+
+const props =
+{
+  match:
+  {
+    params: {
+      id: '5df8d913-79b9-42fd-a17b-ffc917add446'
```
Comment (wesruv) [inline pantheon-bundle/frontend/src/app/productDetails.test.tsx:41]: What's this ID, not sure if it's debug code? If it isn't, maybe a comment to explain why there's a hardcoded id?

File: pantheon-bundle/frontend/src/app/contexts/ProductContext.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useState, useEffect } from "react";
+
+export let ProductContext = createContext<IProduct[]>([] as IProduct[]);
```
Comment (grantsydney) [inline pantheon-bundle/frontend/src/app/contexts/ProductContext.tsx:3]: `ProductContext` was created as a way to share `allProducts` between `ProductDetails` and `ProductListing`

File: pantheon-bundle/frontend/src/app/productListing.tsx [hunk 3]
```diff
@@ -1,188 +1,119 @@
...
-          });
-        }
-      })
-  }
-
-  private getProductsUrl() {
-    const backend = "/content/products.query.json?nodeType=pant:product&orderby=name"
-    return backend
-  }
+  const onToggle = (id) => {
```
Comment (wesruv) [inline pantheon-bundle/frontend/src/app/productListing.tsx:20]: I can has comment for function?
e.g. 
```js
/**
 * Does effects for things
 * @param [string] id Description on id
 */
```

File: pantheon-bundle/frontend/src/app/productListing.tsx [hunk 4]
```diff
@@ -1,188 +1,119 @@
...
+  };
 
-  private onToggle = (id) => (event: any) => {
-    this.state.results.map(data => {
-      if (data["jcr:uuid"] === id) {
-        (data.isOpen as any) = !data.isOpen
-        this.setState({ isProductDetails: false })
-      }
-    });
+  const onSelect = (id) => {
```
Comment (wesruv) [inline pantheon-bundle/frontend/src/app/productListing.tsx:24]: I can has comment for function?

File: pantheon-bundle/frontend/src/app/productListing.tsx [hunk 5]
```diff
@@ -1,188 +1,119 @@
...
+    // @ts-ignore
+    window.location += `/${id}`;
   };
 
-  private onSelect = (event, data) => () => {
-    this.setState({
-      isProductDetails: !this.state.isProductDetails,
-      productName: data.name,
+  const setInput = (text, event) => {
```
Comment (wesruv) [inline pantheon-bundle/frontend/src/app/productListing.tsx:29]: I can has comment for function?

File: pantheon-bundle/frontend/src/app/productListing.tsx [hunk 6]
```diff
@@ -1,188 +1,119 @@
...
 
-  private setInput = input => {
-    const versions: string[] = [];
-    let searchString = "";
-    this.setState({ input })
-    this.state.allProducts.map(data => {
-      searchString = "" + data.name
-      if (searchString.toLowerCase().includes(input.toLowerCase())) {
-        versions.push(data)
+  function updateProduct(id: string) {
```
Comment (wesruv) [inline pantheon-bundle/frontend/src/app/productListing.tsx:37]: I can has comment for function?

**Review by wesruv — COMMENTED** [review]
Code looks good to me, @xdavidson can you test it before we approve?

File: pantheon-bundle/frontend/src/app/contexts/ProductContext.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+export interface IProductProviderProps {
+  children?: React.ReactNode;
+}
+
+export interface IProduct {
+  description: string,
+  isOpen: boolean,
+  ["jcr:created"]: number,
```
Comment (xdavidson) [inline pantheon-bundle/frontend/src/app/contexts/ProductContext.tsx:12]: "jcr:created" if a type Date. e.g.
"jcr:created": "Mon Sep 28 2020 11:20:19 GMT+0000",

File: pantheon-bundle/frontend/src/app/contexts/ProductContext.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  ["jcr:createdBy"]: string,
+  ['jcr:lastModified']: number,
```
Comment (xdavidson) [inline pantheon-bundle/frontend/src/app/contexts/ProductContext.tsx:14]: "jcr:lastModified" is a type Date. e.g
"jcr:lastModified": "Mon Sep 28 2020 11:20:19 GMT+0000",

File: pantheon-bundle/frontend/src/app/productListing.tsx (hunk 3 above)
Comment (wesruv) [reply pantheon-bundle/frontend/src/app/productListing.tsx:24]: It's beautiful 😢

File: pantheon-bundle/frontend/src/app/productListing.tsx (hunk 5 above)
Comment (wesruv) [reply pantheon-bundle/frontend/src/app/productListing.tsx]: Does this one need a comment? I vote yes, but if you feel like it doesn't I'm cool with it :D

**Comment by xdavidson** [comment]
I tested this branch locally. The backbutton works when I go from product details to product Listing.

File: pantheon-bundle/frontend/src/app/productListing.tsx (hunk 5 above)
Comment (grantsydney) [reply pantheon-bundle/frontend/src/app/productListing.tsx]: yes added a comment to this function in a diff commit

**Review by xdavidson — APPROVED** [review]

**Pull request merged**